docs/community/tutorials-contest/2022/erc20-as-subnet-gas-token/README.md

```diff
@@ -0,0 +1,468 @@
+
+### Lock-mint and Burn-release Mechanisms
+
+Reason we have chosen these mechanisms is because our ERC-20 token might not be mintable and we are sure that our Subnet's gas token is mintable.
+
+#### How Lock-mint Works:
+
+- User deposits tokens to the bridge contract, effectively locking it.
+- Relayer detects the transfer event and sends a transaction to the bridge contract that is on the other chain.
+- Bridge contract on the other chain mints the token and sends it to the given address.
+
+We are using the Lock-Mint mechanism when we bridge our token from C-chain to Subnet.
+
+#### How Burn-release Works:
+
+- User deposits tokens to the bridge contract, and the bridge contract sends them to the 0 address, effectively burning it.
+- Relayer detects the transfer event and sends a transaction to the bridge contract that is on the other chain.
+- Bridge contract on the other chain sends the token to the given address, effectively releasing it.
+
+We are using the Burn-Release mechanism when we bridge our token from Subnet to C-chain.
+
+:::caution
+
+If bridge contract does not have enough tokens, it can not release them. Therefore, make sure that bridge is sufficiently funded to release intended amount. This was not a concern for lock-mint mechanism because bridge could always mint.
+
+:::
+
+### Building Blocks of Our Bridge
+
+#### Relayer
+
+Relayer is an off-chain application that listens for events on both chains. Upon events it sends transactions to other chain's bridge contracts. It has the private key of the bridge contracts' admin account allowing it to mint or release tokens.
+
+#### Contracts
+
+We will have bridge contracts on both chains. Users will send transactions to these contracts when they want to burn or lock their token on the respective chain. When a burn or lock happens these contracts emit an event for the relayer to observe. When the relayer observes the event on one chain, it will call bridge contract on the other chain with either mint or release function.
+
+## Requirements
+
+- [NodeJS](https://nodejs.org/en/) must be installed.
+- [npm](https://www.npmjs.com/) must be installed.
+- [Hardhat](https://www.npmjs.com/) must be installed.
+
+## Getting Started
+
+## Initialize the Project
+
+Let's start by initializing our workspace with [Hardhat](https://hardhat.org/).
+
+To initialize project, run:
+
+```bash
+npx hardhat init
+```
+
+Select `Create a JavaScript project` and walk through creating the project.
+
+## Create Contracts
+
+Delete `Lock.sol` file
+
+To use openzeppelin contracts in our contracts, run:
+
+```bash
+npm i @openzeppelin/contracts
+```
+
+### Create Token Contracts
+
+Create a `Token` folder inside `contracts` folder
+
+#### Avax Token
+
+Create `AvaxToken.sol` file inside `Token` folder
```

Comment:
I want to make sure I understand you correctly. You are suggesting I upload AvaxToken.sol rather than AvaxToken.sol.md and link AvaxToken.sol inside the tutorial right? If I understood you correctly, I have done the change you have suggested and ran it locally. It seems worse for me therefore, I wanted to make sure I am doing the right thing. I am sharing you a video of me running it locally and using both AvaxToken.sol and AvaxToken.sol.md. I am looking forward for your feedback

https://user-images.githubusercontent.com/65618011/181860125-1ff112dd-9f4c-4fba-bcf8-216796e878cf.mov

.

Reply:
Hmm. It looks like it's not rendering well. You can keep the same version since I want to generate the same thing without any complexity of parsing code files to markdown—so no need to change anything here 🙂